profiles/lib/handlers.lua

```diff
@@ -399,15 +399,22 @@ function Handlers.handle_oneway(way,result,data,profile)
   end
 end
 
+function Handlers.handle_weights(way,result,data,profile)
+  if properties.weight_name == 'distance' then
+    -- set weight rates to 1 for the distance weight, edge weights are distance / rate
+    result.forward_rate = 1
```

Comment:
but why are forward and backward rates still being set, without checking whether the way is accessible (mode != 0 and speed > 0)? if the way is not accessible, i don't think the rate should be set.